File: Desktop/intern 4th yr DSA/circularqueue.py
size = int(input("Enter size: "))
q = [None] * size
front = -1
rear = -1

while True:
    print("\n1.Insert")
    print("2.Delete")
    print("3.Display")
    print("4.Exit")

    ch = int(input("Enter choice: "))

    if ch == 1:
        x = int(input("Enter element: "))

        if (rear + 1) % size == front:
            print("Overflow")
        else:
            if front == -1:
                front = rear = 0
            else:
                rear = (rear + 1) % size

            q[rear] = x

    elif ch == 2:
        if front == -1:
            print("Underflow")
        else:
            print("Deleted:", q[front])

            if front == rear:
                front = rear = -1
            else:
                front = (front + 1) % size

    elif ch == 3:
        if front == -1:
            print("Queue is empty")
        else:
            i = front

            while True:
                print(q[i], end=" ")
                if i == rear:
                    break
                i = (i + 1) % size

            print()

    elif ch == 4:
        break

    else:
        print("Invalid choice")

File: Day3/circularQueue.cpp
#include <iostream>
using namespace std;

const int SIZE = 5;

int cq[SIZE];
int front = -1;
int rear = -1;

void enqueue(int x) {
    if ((rear + 1) % SIZE == front) {
        cout << "Queue Overflow\n";
        return;
    }
    if (front == -1) { 
        front = rear = 0;
    } else {
        rear = (rear + 1) % SIZE;
    }
    cq[rear] = x;
}

void dequeue() {
    if (front == -1) {
        cout << "Queue Underflow\n";
        return;
    }
    cout << "Deleted: " << cq[front] << endl;
    if (front == rear) {
        front = rear = -1;
    } else {
        front = (front + 1) % SIZE;
    }
}

void display() {
    if (front == -1) {
        cout << "Queue Empty\n";
        return;
    }
    int i = front;
    while (true) {
        cout << cq[i] << " ";
        if (i == rear)
            break;
        i = (i + 1) % SIZE;
    }
    cout << endl;
}

int main() {
    enqueue(10);
    enqueue(20);
    enqueue(30);
    enqueue(40);
    display();
    dequeue();
    dequeue();
    display();
    enqueue(50);
    enqueue(60);
    display();
    return 0;
}